pkg/bootstrap/node.go

```diff
@@ -93,13 +93,59 @@ func GetPoolSize(baseImageName string, nodes int) (int64, error) {
 	return poolSize, nil
 }
 
+func setPoolLimit(poolSize int64) error {
+	var cmdPath string
+	var err error
+
+	if cmdPath, err = exec.LookPath("machinectl"); err != nil {
+		return fmt.Errorf("machinectl not installed: %s", err)
+	}
+
+	args := []string{
+		cmdPath,
+		"set-limit",
+		strconv.FormatInt(poolSize, 10),
+	}
+
+	cmd := exec.Cmd{
+		Path:   cmdPath,
+		Args:   args,
+		Env:    os.Environ(),
+		Stdout: os.Stdout,
+		Stderr: os.Stderr,
+	}
+
+	if err := cmd.Run(); err != nil {
+		// We do expect this might occur. E.g. if they have
+		// already started using the pool
+		return fmt.Errorf("error running machinectl: %s", err)
+	}
+
+	return nil
+}
+
 func EnlargeStoragePool(poolSize int64) error {
-	// It is equivalent to the following shell commands:
+	// Check to see if the filesystem is already >= poolSize
+	var stat syscall.Statfs_t
+	if err := syscall.Statfs(machinesDir, &stat); err == nil {
+		if stat.Bsize >= poolSize {
+			return nil
+		}
+	}
+
+	// First call `machinectl set-limit size`. If this succeeds,
+	// we are done
+	if err := setPoolLimit(poolSize); err == nil {
+		return nil
```

Comment:
Usually Go's error handling is done the other way around, like:

```
if err != nil {
        return err
}
```

I feel awkward whenever I see the opposite way of error handling.
I know it's sometimes hard to handle errors in a correct way, and I'm not sure if we should always force everyone to write in that way.
So I think that the code is fine for now.
But in the future, it would be really great, if you could follow the idiomatic coding style. :-)

Reply:
in this case its not checking for error, its checking for success. E.g. if setPoolLimit() succeeds, we are done. Only if it fails do we need to try the backup 'unmount/resize/remount'.

e.g. i would not want to return err @ this time.